internal/service/resiliencehub/app.go

```diff
@@ -0,0 +1,1502 @@
+	if err != nil {
+		// If draft doesn't exist, try published version
+		templateInput.AppVersion = aws.String("release")
+		templateOutput, err = conn.DescribeAppVersionTemplate(ctx, templateInput)
+	}
+
+	if err != nil {
+		// Template doesn't exist, set to null
+		state.AppTemplate = types.ListNull(types.ObjectType{
+			AttrTypes: map[string]attr.Type{
+				names.AttrVersion: types.StringType,
+				"additional_info": types.MapType{ElemType: types.StringType},
+				"app_component":   types.ListType{ElemType: appComponentObjectType()},
+				"resource":        types.ListType{ElemType: resourceObjectType()},
+			},
+		})
+	} else {
+		// Flatten the template
+		appTemplate, diags := r.flattenAppTemplate(ctx, *templateOutput.AppTemplateBody)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		state.AppTemplate = appTemplate
+	}
+
+	// Read input sources (resource mappings)
+	// Try both draft and release versions since import might be asynchronous
+	var inputSourcesOutput *resiliencehub.ListAppInputSourcesOutput
+	var inputSourcesErr error
+
+	// First try draft version
+	inputSourcesInput := &resiliencehub.ListAppInputSourcesInput{
+		AppArn:     state.ID.ValueStringPointer(),
+		AppVersion: aws.String("draft"),
+	}
+
+	inputSourcesOutput, inputSourcesErr = conn.ListAppInputSources(ctx, inputSourcesInput)
+
+	if inputSourcesErr != nil || inputSourcesOutput == nil || len(inputSourcesOutput.AppInputSources) == 0 {
+		// Try release version
+		inputSourcesInput.AppVersion = aws.String("release")
+		inputSourcesOutput, inputSourcesErr = conn.ListAppInputSources(ctx, inputSourcesInput)
+	}
+
+	if inputSourcesErr != nil || inputSourcesOutput == nil || len(inputSourcesOutput.AppInputSources) == 0 {
+		// Input sources might not be immediately available after app creation/update
+		// Preserve existing resource mappings from state to prevent unnecessary drift
+		if state.ResourceMapping.IsNull() || state.ResourceMapping.IsUnknown() {
+			// No previous state exists, set to null
+			state.ResourceMapping = types.ListNull(resourceMappingObjectType())
+		}
+	} else {
+		// Parse the app template JSON for resource mappings
+		var appTemplateMap map[string]any
+		if err := json.Unmarshal([]byte(*templateOutput.AppTemplateBody), &appTemplateMap); err != nil {
+			resp.Diagnostics.AddError("Failed to parse app template", err.Error())
+			return
+		}
+
+		// Flatten the resource mappings
+		resourceMappings, diags := r.flattenResourceMappings(ctx, inputSourcesOutput.AppInputSources, appTemplateMap)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		state.ResourceMapping = resourceMappings
+	}
+
+	// Handle tags - use listTags to get all tags including provider tags
+	tags, err := listTags(ctx, conn, state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to list tags", err.Error())
+		return
+	}
+	setTagsOut(ctx, tags.Map())
```

Comment:
Since there is a `Tags` field on the `awstypes.App` returned by `FindAppByARN`, a separate `listTags` call is not needed, instead use the following:

```suggestion
	setTagsOut(ctx, output.Map())
```